blockly-ws-search/workspace_search.js

```diff
@@ -395,4 +401,62 @@ class WorkspaceSearch {
       Blockly.utils.dom.removeClass(blockPath, 'searchHighlight');
     });
   }
+
+  /**
+   * Scrolls workspace to bring given block into view.
+   * @param {Blockly.BlockSvg} block Block to bring into view.
+   * @private
+   */
+  scrollToVisible_(block) {
+    if (!this.workspace_.isMovable()) {
+      console.warn('Cannot scroll to block in a non-movable' +
+          'workspace.');
+      return;
+    }
+    // XY is in workspace coordinates.
+    const xy = block.getRelativeToSurfaceXY();
+    const scale = this.workspace_.scale;
+
+    // Block bounds in pixels relative to the workspace origin (0,0 is centre).
+    const width = block.width * scale;
+    const height = block.height * scale;
+    const top = xy.y * scale;
+    const bottom = (xy.y + block.height) * scale;
+    // In RTL the block's position is the top right of the block, not top left.
+    const left = this.workspace_.RTL ? xy.x * scale - width: xy.x * scale;
+    const right = this.workspace_.RTL ? xy.x * scale : xy.x * scale +  width;
+
```

Comment:
Looks great : ) I would maybe make it blockTop, blockBottom etc. just for readability sake